lib/json/serialization.nit

```diff
@@ -448,6 +448,26 @@ class JsonDeserializer
 	end
 end
 
+redef class Text
+
+	# Deserialize a `nullable Object` from this JSON formatted string
+	#
+	# Warning: Deserialization errors are reported with `print_error` and
+	# may be returned as a partial object or as `null`.
+	#
+	# This method is not appropriate when errors need to be handled programmatically,
+	# manually use a `JsonDeserializer` in such cases.
+	fun from_json_string: nullable Object
+	do
+		var deserializer = new JsonDeserializer(self)
+		var res = deserializer.deserialize
+		if deserializer.errors.not_empty then
+			print_error "Deserialization Errors: {deserializer.errors.join(", ")}"
```

Comment:
Do we really need to distinguish between deserializing an Error object and a deserialisation error?
As you said dirty script wont care anyway, and those wo care will use the full API.
However, I do think that not-so-dirty script will be happy to handle their requests with no rogue print and simple way to catch most errors.